neural_compressor/torch/algorithms/weight_only/autoround.py

```diff
@@ -12,13 +12,169 @@
+        """
+
+        self.model = model
+        self.tokenizer = None
+        self.weight_config = weight_config
+        self.enable_full_range = enable_full_range
+        self.batch_size = batch_size
+        self.amp = amp
+        self.device = device
+        self.lr_scheduler = lr_scheduler
+        self.use_quant_input = use_quant_input
+        self.enable_minmax_tuning = enable_minmax_tuning
+        self.lr = lr
+        self.minmax_lr = minmax_lr
+        self.low_gpu_mem_usage = low_gpu_mem_usage
+        self.iters = iters
+        self.seqlen = seqlen
+        self.n_samples = n_samples
+        self.sampler = sampler
+        self.seed = seed
+        self.n_blocks = n_blocks
+        self.gradient_accumulate_steps = gradient_accumulate_steps
+        self.not_use_best_mse = not_use_best_mse
+        self.dynamic_max_gap = dynamic_max_gap
+        self.data_type = "int"
+        self.scale_dtype = scale_dtype
+
+    def quantize(self, model: torch.nn.Module, *args, **kwargs):
+        run_fn = kwargs.get("run_fn", None)
+        run_args = kwargs.get("run_args", None)
+        assert run_fn is not None, (
+            "Can't find run_func. Please provide run_func to quantize API "
+            "or overwrite quantize member function in your Quantizer class."
+        )
+        model = self.prepare(model)
+        if run_args:
+            run_fn(model, *run_args)
+        else:
+            run_fn(model)
+        model = self.convert(model)
+        return model
+
+    def prepare(self, model: torch.nn.Module, *args, **kwargs):
+        """Prepares a given model for quantization.
+        Args:
+            model (torch.nn.Module): The model to be prepared.
+
+        Returns:
+            A prepared model.
+        """
+        self.rounder = AutoRoundProcessor(
+            model=model,
+            tokenizer=None,
+            weight_config=self.weight_config,
+            enable_full_range=self.enable_full_range,
+            batch_size=self.batch_size,
+            amp=self.amp,
+            device=self.device,
+            lr_scheduler=self.lr_scheduler,
+            use_quant_input=self.use_quant_input,
+            enable_minmax_tuning=self.enable_minmax_tuning,
+            lr=self.lr,
+            minmax_lr=self.minmax_lr,
+            low_gpu_mem_usage=self.low_gpu_mem_usage,
+            iters=self.iters,
+            seqlen=self.seqlen,
+            n_samples=self.n_samples,
+            sampler=self.sampler,
+            seed=self.seed,
+            n_blocks=self.n_blocks,
+            gradient_accumulate_steps=self.gradient_accumulate_steps,
+            not_use_best_mse=self.not_use_best_mse,
+            dynamic_max_gap=self.dynamic_max_gap,
+            data_type=self.data_type,
+            scale_dtype=self.scale_dtype,
+        )
+        self.rounder.prepare()
+        return model
+
+    def convert(self, model: torch.nn.Module, *args, **kwargs):
+        model, weight_config = self.rounder.convert()
```

Comment:
If mode=Mode.CONVERT, `self.rounder` will not be found, because only the `convert` function will be executed in this case. Maybe you should move `self.rounder` to `__init__`?

Reply:
Per offline synced, self.rounder initialization is placed in `prepare`.